File: classic/cs/tirepd/B2G/B2GException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace B2G
{
    public class B2GException : ApplicationException
    {
        private AckReturnCode returnCode;
        public AckReturnCode ReturnCode { get { return returnCode; } }
        public B2GException(AckReturnCode returnCode)
            : base("B2G Error")
        {
            this.returnCode = returnCode;
        }
        public B2GException(AckReturnCode returnCode, Exception ex)
            : base("B2G Error", ex)
        {
            this.returnCode = returnCode;
        }
        public B2GException(string errorMessage, AckReturnCode returnCode)
            : base(errorMessage)
        {
            this.returnCode = returnCode;
        }
        public B2GException(string errorMessage, AckReturnCode returnCode, Exception ex)
            : base(errorMessage, ex)
        {
            this.returnCode = returnCode;
        }
    }
}

File: classic/vb/tirepd/VB.B2G/B2GException.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace B2G
    Public Class B2GException
        Inherits ApplicationException

        Private _returnCode As AckReturnCode

        Public ReadOnly Property ReturnCode As AckReturnCode
            Get
                Return Me._returnCode
            End Get
        End Property

        Public Sub New(ByVal returnCode As AckReturnCode)
            MyBase.New("B2G Error")
            Me._returnCode = returnCode
        End Sub

        Public Sub New(ByVal returnCode As AckReturnCode, ByVal ex As Exception)
            MyBase.New("B2G Error", ex)
            Me._returnCode = returnCode
        End Sub

        Public Sub New(ByVal errorMessage As String, ByVal returnCode As AckReturnCode)
            MyBase.New(errorMessage)
            Me._returnCode = returnCode
        End Sub

        Public Sub New(ByVal errorMessage As String, ByVal returnCode As AckReturnCode, ByVal ex As Exception)
            MyBase.New(errorMessage, ex)
            Me._returnCode = returnCode
        End Sub
    End Class
End Namespace
